library/ccm.c

```diff
@@ -35,6 +35,10 @@
 #include "mbedtls/error.h"
 #include "mbedtls/constant_time.h"
 
+#if !defined(MBEDTLS_CIPHER_C)
```

Comment:
I would prefer `MBEDTLS_CIPHER_LIGHT` here instead because it seems more precise to me.
The same also for the GCM module

Reply:
It's probably better for readability, but from a purely logical perspective once we know `CCM_C` (or `GCM_C`) is defined, then `!CIPHER_C` and `CIPHER_LIGHT` are equivalent.